js/ui/tree_view/ui.tree_view.base.js

```diff
@@ -861,21 +878,53 @@ const TreeViewBase = HierarchicalCollectionWidget.inherit({
         $icon.hide();
     },
 
-    _renderToggleItemVisibilityIcon: function($node, node) {
-        const $icon = $('<div>')
-            .addClass(TOGGLE_ITEM_VISIBILITY_CLASS)
-            .appendTo($node);
+    _renderToggleItemVisibilityCustomIcon: function($node, node) {
+        const $expandIcon = getImageContainer(this.option('expandButtonIcon') ?? this.option('collapseButtonIcon'));
+        const $collapseIcon = getImageContainer(this.option('collapseButtonIcon') ?? this.option('expandButtonIcon'));
+
+        $expandIcon.appendTo($node).addClass('dx-custom-expander-icon');
+        $collapseIcon.appendTo($node).addClass('dx-custom-expander-icon');
+
+        $expandIcon.addClass(CUSTOM_ICON_TOGGLE_ITEM_VISIBILITY_CLASS);
```

Comment:
There are too many the same actions with expandIcon and collapseIcon:
1. appendTo same container
2. add same classes
3. then same classes for disabled state
4.  then call _renderToggleItemVisibilityIconClick method for both

Can we all of this to move in extra method and call it with each icon separatly?